File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.py
# [def(int)] dp
# : Function that updates number of execution: dCount
# and returns integer of n-th fibonacci
#
# - Parameters
# [int] n: Integer that represents number for finding nth fibonacci
#
# - Return
# [int] fib[n]: Integer of nth fibonacci
def dp(n: int):
    global dCount

    # [list] fib: List of "nth fibonacci" at each index
    fib = [0] * (n + 1)
    
    # First and Second numbers are 1
    fib[1] = fib[2] = 1
    
    # Repeat 3 through n to update fib
    for i in range(3, n+1):
        # Updates dCount
        dCount += 1
        # Use pre-caculated values
        fib[i] = fib[i-1] + fib[i-2]
    
    # Return fib[n] that number of nth fibonacci
    return fib[n]

# [int] n: Integer that represents number for finding nth fibonacci
n = int(input())

dCount = 0

# Number of recursive function executions, rCount is 
# same to nth fibonacci
rCount = dp(n)

print(f"{rCount} {dCount}")

File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.swift
// [Int] n (immutable)
// : Integer that represents number for finding nth fibonacci
if let input: String = readLine(), let n: Int = Int(input) {
    // [Int] count (mutable)
    // : Integer to store the number of executions of dp function
    var dCount: Int = 0
    
    // [Array<Int>] fib (mutable)
    // : List of Integer to store the nth fibonacci
    var fib = [Int](repeating: 0, count: n+1)
    
    // First and Second numbers are 1
    fib[1] = 1
    fib[2] = 1
    
    // Repeat 3 through n to update fib
    for i in 3...n {
        // Updates dCount
        dCount += 1
        
        // Use pre-caculated values
        fib[i] = fib[i-1] + fib[i-2]
    }
    
    // Print a number of nth fibonacci and execution time of dp
    print("\(fib[n]) \(dCount)")
}